diff --git a/hip-config.cmake.in b/hip-config.cmake.in
index 89d1224e..b4f5a797 100755
--- a/hip-config.cmake.in
+++ b/hip-config.cmake.in
@@ -300,6 +300,7 @@ if(HIP_COMPILER STREQUAL "clang")
       ${HIP_CLANGRT_LIB_SEARCH_PATHS}
       ${HIP_CLANG_INCLUDE_PATH}/../lib/linux
       ${HIP_CLANG_INCLUDE_PATH}/../lib/windows
+      GUIX_CLANG_RUNTIME_DIR/lib/linux
       NO_DEFAULT_PATH)
 
   # Add support for __fp16 and _Float16, explicitly link with compiler-rt
diff --git a/src/hip_intercept.cpp b/src/hip_intercept.cpp
index e4e7e769..972ae7a3 100644
--- a/src/hip_intercept.cpp
+++ b/src/hip_intercept.cpp
@@ -47,7 +47,7 @@ const char* hipKernelNameRefByPtr(const void* host_function, hipStream_t stream)
                                     : nullptr;
 }
 
-void hipRegisterTracerCallback(const void* function) {
+void hipRegisterTracerCallback(void* function) {
   activity_prof::report_activity.store(
       reinterpret_cast<decltype(activity_prof::report_activity.load())>(function),
       std::memory_order_relaxed);
