diff --git a/hipamd/CMakeLists.txt b/hipamd/CMakeLists.txt
index c14a9ad32..f882e890b 100755
--- a/hipamd/CMakeLists.txt
+++ b/hipamd/CMakeLists.txt
@@ -400,8 +400,8 @@ if (NOT ${HIPCC_BIN_DIR} STREQUAL "")
   install(PROGRAMS ${HIPCC_BIN_DIR}/hipcc.pl DESTINATION bin)
   install(PROGRAMS ${HIPCC_BIN_DIR}/hipconfig.pl DESTINATION bin)
   install(PROGRAMS ${HIPCC_BIN_DIR}/hipvars.pm DESTINATION bin)
-  install(PROGRAMS ${HIPCC_BIN_DIR}/hipcc.bat DESTINATION bin)
-  install(PROGRAMS ${HIPCC_BIN_DIR}/hipconfig.bat DESTINATION bin)
+  # install(PROGRAMS ${HIPCC_BIN_DIR}/hipcc.bat DESTINATION bin)
+  # install(PROGRAMS ${HIPCC_BIN_DIR}/hipconfig.bat DESTINATION bin)
 endif()
 
 #############################
