diff --git a/clang/CMakeLists.txt b/clang/CMakeLists.txt
index 1df7a757dc74..bc99e5bba897 100644
--- a/clang/CMakeLists.txt
+++ b/clang/CMakeLists.txt
@@ -120,6 +120,8 @@ if(CLANG_BUILT_STANDALONE)
       endif()
     endif()
 
+    add_subdirectory(${LLVM_THIRD_PARTY_DIR}/unittest third-party/unittest)
+
     if (NOT TARGET llvm_gtest)
         message(FATAL_ERROR "llvm-gtest not found. Please install llvm-gtest or disable tests with -DLLVM_INCLUDE_TESTS=OFF")
     endif()
