From a44bb42342ca86d25ee9d034011049e821ef501c Mon Sep 17 00:00:00 2001
From: Thomas Gibson <14180421+thomasgibson@users.noreply.github.com>
Date: Thu, 27 Oct 2022 17:38:30 -0500
Subject: [PATCH] Patch HPCG CMake

---
 CMakeLists.txt           | 12 ++++++------
 cmake/Dependencies.cmake |  2 +-
 2 files changed, 7 insertions(+), 7 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 236bc46..dc1f6be 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -74,12 +74,12 @@ include(cmake/Dependencies.cmake)
 
 # Availability of rocm_check_target_ids command assures that we can also build
 # for gfx90a target
-if(COMMAND rocm_check_target_ids)
-    set(DEFAULT_AMDGPU_TARGETS "gfx900:xnack-;gfx906:xnack-;gfx908:xnack-;gfx908:xnack+;gfx90a:xnack-;gfx90a:xnack+")
-else()
-    set(DEFAULT_AMDGPU_TARGETS "gfx900:xnack-;gfx906:xnack-;gfx908:xnack-;gfx908:xnack+")
-endif()
-set(AMDGPU_TARGETS "${DEFAULT_AMDGPU_TARGETS}" CACHE STRING "List of specific machine types for library to target")
+# if(COMMAND rocm_check_target_ids)
+#     set(DEFAULT_AMDGPU_TARGETS "gfx900:xnack-;gfx906:xnack-;gfx908:xnack-;gfx908:xnack+;gfx90a:xnack-;gfx90a:xnack+")
+# else()
+#     set(DEFAULT_AMDGPU_TARGETS "gfx900:xnack-;gfx906:xnack-;gfx908:xnack-;gfx908:xnack+")
+# endif()
+# set(AMDGPU_TARGETS "${DEFAULT_AMDGPU_TARGETS}" CACHE STRING "List of specific machine types for library to target")
 
 # Find HIP package
 find_package(HIP REQUIRED)
diff --git a/cmake/Dependencies.cmake b/cmake/Dependencies.cmake
index 22259f4..7b490e0 100644
--- a/cmake/Dependencies.cmake
+++ b/cmake/Dependencies.cmake
@@ -26,7 +26,7 @@
 # Dependencies
 
 # Git
-find_package(Git REQUIRED)
+find_package(Git)
 
 # Add some paths
 list(APPEND CMAKE_PREFIX_PATH ${ROCM_PATH} ${ROCM_PATH}/hip)
-- 
2.25.1

