Remove distro and FHS specific logic

diff --git a/clang/lib/Driver/ToolChains/AMDGPU.cpp b/clang/lib/Driver/ToolChains/AMDGPU.cpp
index 88d22f7149d2..f0ac8658becc 100644
--- a/clang/lib/Driver/ToolChains/AMDGPU.cpp
+++ b/clang/lib/Driver/ToolChains/AMDGPU.cpp
@@ -269,7 +269,7 @@ RocmInstallationDetector::getInstallationPathCandidates() {
     ROCmSearchDirs.emplace_back(RealClangRoot.str(), /*StrictChecking=*/true);
   ROCmSearchDirs.emplace_back(D.ResourceDir,
                               /*StrictChecking=*/true);
-
+  #if 0
   ROCmSearchDirs.emplace_back(D.SysRoot + "/opt/rocm",
                               /*StrictChecking=*/true);

@@ -308,6 +308,7 @@ RocmInstallationDetector::getInstallationPathCandidates() {
   if (!LatestROCm.empty())
     ROCmSearchDirs.emplace_back(D.SysRoot + "/opt/" + LatestROCm,
                                 /*StrictChecking=*/true);
+   #endif

   DoPrintROCmSearchDirs();
   return ROCmSearchDirs;
diff --git a/openmp/libomptarget/plugins/amdgpu/impl/impl.cpp b/openmp/libomptarget/plugins/amdgpu/impl/impl.cpp
index 80e024789b64..3a14e0889ded 100644
--- a/openmp/libomptarget/plugins/amdgpu/impl/impl.cpp
+++ b/openmp/libomptarget/plugins/amdgpu/impl/impl.cpp
@@ -21,10 +21,11 @@ bool is_locked(void *ptr, hsa_status_t *err_p, void **agentBaseAddress) {
   info.size = sizeof(hsa_amd_pointer_info_t);
   err = hsa_amd_pointer_info(ptr, &info, nullptr, nullptr, nullptr);

-  if (err != HSA_STATUS_SUCCESS)
+  if (err != HSA_STATUS_SUCCESS) {
     DP("Error when getting pointer info\n");
-  else
+  } else {
     is_locked = (info.type == HSA_EXT_POINTER_TYPE_LOCKED);
+  }

   if (is_locked && agentBaseAddress != nullptr) {
     // When user passes in a basePtr+offset we need to fix the
diff --git a/openmp/libomptarget/src/CMakeLists.txt b/openmp/libomptarget/src/CMakeLists.txt
index a40bb4dffd35..90fd6b495d1a 100644
--- a/openmp/libomptarget/src/CMakeLists.txt
+++ b/openmp/libomptarget/src/CMakeLists.txt
@@ -64,7 +64,8 @@ target_link_libraries(omptarget PRIVATE
   ${CMAKE_DL_LIBS}
   ${LLVM_OFFLOAD_ARCH}
   "-Wl,--no-allow-shlib-undefined"
-  "-Wl,--version-script=${CMAKE_CURRENT_SOURCE_DIR}/exports")
+  # "-Wl,--version-script=${CMAKE_CURRENT_SOURCE_DIR}/exports"
+  )

 # libomptarget.so needs to be aware of where the plugins live as they
 # are now separated in the build directory.
