Minor tweaks to avoid compilation errors of openmp/libomptarget

diff --git a/clang/lib/Driver/ToolChains/AMDGPU.cpp b/clang/lib/Driver/ToolChains/AMDGPU.cpp
index fe3664647560..bab3b75b6314 100644
--- a/clang/lib/Driver/ToolChains/AMDGPU.cpp
+++ b/clang/lib/Driver/ToolChains/AMDGPU.cpp
@@ -271,6 +271,7 @@ RocmInstallationDetector::getInstallationPathCandidates() {
   ROCmSearchDirs.emplace_back(D.ResourceDir,
                               /*StrictChecking=*/true);

+  #if 0
   ROCmSearchDirs.emplace_back(D.SysRoot + "/opt/rocm",
                               /*StrictChecking=*/true);

@@ -318,6 +319,7 @@ RocmInstallationDetector::getInstallationPathCandidates() {
                                 /*StrictChecking=*/true);
   }

+  #endif
   DoPrintROCmSearchDirs();
   return ROCmSearchDirs;
 }
