diff --git a/CMakeLists.txt b/CMakeLists.txt
index b83b2f4..62695fb 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -89,7 +89,7 @@ if(HIP_VERSION VERSION_GREATER_EQUAL "4.3")
   set(AMDGPU_TARGETS "${AMDGPU_TARGETS};gfx90a")
 endif()
 if (HIP_VERSION VERSION_GREATER_EQUAL "5.3")
-  set(AMDGPU_TARGETS "${AMDGPU_TARGETS};gfx940")
+  set(AMDGPU_TARGETS "${AMDGPU_TARGETS};gfx940;gfx1030")
 endif()
 if (HIP_VERSION VERSION_GREATER_EQUAL "5.7")
   set(AMDGPU_TARGETS "${AMDGPU_TARGETS};gfx941;gfx942")
